keyboardClass.js

```diff
@@ -0,0 +1,241 @@
+        this.targetOfKeyboard.addEventListener('blur', () => {
+            this.posIn = this.targetOfKeyboard.selectionStart;
+        });
+        document.addEventListener('keyup', (event) => {
+            const keyCode = event.code ? event.code : event.keyCode;
+            if (!this.btnList[keyCode]) return;
+            if (keyCode === 'ShiftLeft' || keyCode === 'ShiftRight') {
+                this.shiftPressed = false;
+                this.changeCase();
+                if (keyCode === 'ShiftLeft') this.btnList.ShiftRight.locked = false;
+                else this.btnList.ShiftLeft.locked = false;
+            } else if (keyCode === 'ControlLeft' || keyCode === 'ControlRight') {
+                event.preventDefault();
+                this.ctrlPressed = false;
+            } else if (keyCode === 'AltLeft' || keyCode === 'AltRight') {
+                event.preventDefault();
+                this.altPressed = false;
+            }
+            this.btnList[keyCode].removeClass('active');
+        });
+        this.targetOfKeyboard.addEventListener('click', () => {
+            this.posIn = this.targetOfKeyboard.selectionStart;
+        });
+        this.totalNumberOfKeys = this.btnList.length;
+        this.rowNumber = 0;
+        this.numberOfKeysInRow = 0;
+        this.posIn = 0;
+        this.buttonListArray = [];
+        this.saveKeyboard();
+        this.changeLanguage.bind(this);
+    }
+
+    addBtn(button) {
+        this.totalNumberOfKeys += 1;
+        this.btnList[button.name] = button;
+        const rowsCounterConst = [14, 14, 14, 14, 14];
+        if (this.numberOfKeysInRow < rowsCounterConst[this.rowNumber]) {
+            this.keyboardRows[this.rowNumber].appendChild(button.domElement);
+        } else {
+            this.rowNumber += 1;
+            this.numberOfKeysInRow = 0;
+            this.keyboardRows[this.rowNumber].appendChild(button.domElement);
+        }
+        this.numberOfKeysInRow += 1;
+    }
+
+    triggerKeyboardEvent(el, keyState = 'keydown', keyCode) {
+        this.targetOfKeyboard.focus();
+        this.eventObj = new Event(keyState);
+        this.eventObj.keyCode = keyCode;
+        this.eventObj.which = keyCode;
+        el.dispatchEvent(this.eventObj);
+    }
+
+    valueApp(mod, value_) {
+        let valueBefore = '';
+        let value = value_ === undefined ? '' : value_;
+        let valueAfter = '';
+        if (mod === 'backspace') valueBefore = ''.concat(this.targetOfKeyboard.value).slice(0, this.targetOfKeyboard.selectionStart - 1);
+        else valueBefore = ''.concat(this.targetOfKeyboard.value).slice(0, this.targetOfKeyboard.selectionStart);
+        if (mod === 'delete') valueAfter = ''.concat(this.targetOfKeyboard.value).slice(this.targetOfKeyboard.selectionEnd + 1);
+        else valueAfter = ''.concat(this.targetOfKeyboard.value).slice(this.targetOfKeyboard.selectionEnd);
+        value = valueBefore.concat(value).concat(valueAfter);
+        if (mod === 'none') {
```

Comment:
once again, dry
and why do you use ''.concat(this.targetOfKeyboard.value) ? isn't it the same as just this.targetOfKeyboard.value